lib/cost_report.rb

```diff
@@ -1,15 +1,89 @@
 # frozen_string_literal: true
 
 require "ht_item_overlap"
+require "utils/waypoint"
+require "services"
 
 # Generates reports based on h_share
 class CostReport
-  attr_accessor :organization, :freq_table
+  attr_accessor :organization, :logger, :maxlines, :target_cost
 
-  def initialize(org = nil)
+  def initialize(org = nil, cost: ENV["target_cost"], lines: 1_000_000, logger: Services.logger)
     @organization = org
-    # { org => { 1 org : count, 2 org : count }
-    @freq_table = Hash.new {|hash, key| hash[key] = Hash.new(0) }
+    @target_cost = cost.to_f
```

Comment:
I'm worried this will silently come in as a zero or as something stupid. `Float(n)` will throw an error if `n` is `nil` or can't be parsed into a float.